src/components/home/MyList.js

```diff
@@ -1,92 +1,111 @@
+  const favoriteCollections = collections?.filter((collection) =>
+    favoriteItemsList?.includes(collection?.id),
+  );
+  const likedItemsList = likedItems?.map((entry) => entry.itemId);
+  const likedCollections = collections?.filter((collection) =>
+    likedItemsList?.includes(collection?.id),
+  );
+  const ownCollections = collections?.filter(
+    (collection) => collection?.creator === member?.get('id'),
+  );
+
+  const [tab, setTab] = useState(0);
+
+  const handleChange = (event, newValue) => {
+    setTab(newValue);
+  };
 
   return (
     <div className={classes.root}>
-      <CssBaseline />
-      <Drawer
-        className={classes.drawer}
-        variant="permanent"
-        classes={{
-          paper: classes.drawerPaper,
-        }}
-      >
-        <div className={classes.toolbar} />
-        <List className={classes.link}>
-          {[MY_FAVORITES, MY_UPLOADS, SAVED_COLLECTIONS].map((text) => (
-            <ListItem button key={t(text)} disabled>
-              <ListItemIcon>
-                <StarBorderIcon />
-              </ListItemIcon>
-              <ListItemText primary={text} />
-            </ListItem>
-          ))}
-        </List>
-      </Drawer>
       <main className={classes.content}>
         <Typography variant="h3" align="center" id={TITLE_TEXT_ID}>
           {t('My Collections')}
         </Typography>
-        <Typography variant="h3" className={classes.typographyMargin}>
-          {t('Favorites')}
-        </Typography>
-        <CollectionsGrid collections={collections} isLoading={isLoading} />
+        <AppBar position="static" color="default" className={classes.tabBar}>
+          <Tabs
+            value={tab}
+            onChange={handleChange}
+            variant="fullWidth"
+            indicatorColor="primary"
+            textColor="primary"
+            aria-label="navigation tabs"
+          >
+            <Tab
+              label={t('My Favorites')}
+              icon={<StarBorderIcon />}
+              id={buildMyListNavigationTabId(0)}
+            />
+            <Tab
+              label={t('My Likes')}
+              icon={<FavoriteBorderIcon />}
+              id={buildMyListNavigationTabId(1)}
+            />
+            <Tab
+              label={t('My Publishments')}
+              icon={<PublishIcon />}
+              id={buildMyListNavigationTabId(2)}
+            />
+          </Tabs>
+        </AppBar>
+        <TabPanel value={tab} index={0}>
+          <CollectionsGrid
+            collections={favoriteCollections}
+            isLoading={isLoading}
+          />
+        </TabPanel>
+        <TabPanel value={tab} index={1}>
+          <CollectionsGrid
+            collections={likedCollections}
+            isLoading={isLoading}
+          />
+        </TabPanel>
+        <TabPanel value={tab} index={2}>
+          <CollectionsGrid collections={ownCollections} isLoading={isLoading} />
+        </TabPanel>
```

Comment:
I think you should create new components for each tab. This will split a bit the mutations, and maybe if you have time you can check out `React.lazy` https://reactjs.org/docs/code-splitting.html which can improve loading performance. I think it will fit well with these tabs.

Reply:
I created new components for each tab. For `React.lazy`, I tried with it but it throw an error saying the REACTDOM does not support Suspense at this moment. Maybe I take care of this later.